File: haskell/tests/perminc01.hs
import Permutation

main = do mapM_ (\s -> putStrLn $ show (lehmer s) ++ '\t' : showCycles s)
	   $ takeWhile (< firstOfDegree 6) $ iterate succ identity
	  putStrLn "---"
	  mapM_ (\s -> putStrLn $ show (lehmer s) ++ '\t' : showCycles s)
	   $ map pred $ takeWhile (> identity) $ iterate pred $ firstOfDegree 6

File: cpp/tests/perminc01.cpp
#include <iostream>
#include "Permutation.hpp"
using namespace std;

int main(void) {
 Permutation begin = Permutation::identity();
 Permutation end = Permutation::firstOfDegree(6);
 Permutation p;
 for (p = begin; p < end; p++)
  cout << p.lehmer() << '\t' << string(p) << endl;
 cout << "---" << endl;
 p = end;
 for (p--; p >= begin; p--) {
  cout << p.lehmer() << '\t' << string(p) << endl;
  if (!p) break;
 }
 return 0;
}
